md5sum.cpp

```diff
@@ -0,0 +1,147 @@
+// Compute checksums of files or strings.
+//
+// Copyright (C) 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016, 2017, 2018, 2019 Gregory W. Chicares.
+//
+// This program is free software; you can redistribute it and/or modify
+// it under the terms of the GNU General Public License version 2 as
+// published by the Free Software Foundation.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program; if not, write to the Free Software Foundation,
+// Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301, USA
+//
+// http://savannah.nongnu.org/projects/lmi
+// email: <gchicares@sbcglobal.net>
+// snail: Chicares, 186 Belle Woods Drive, Glastonbury CT 06033, USA
+
+#include "pchfile.hpp"
+
+#include "md5sum.hpp"
+#include "md5.hpp"
+#include "miscellany.hpp"
+#include "assert_lmi.hpp"
+
+#include <fstream>
+
+bool md5sum_vec_from_file
+    (std::string const& filename
+    , std::vector<md5sum_for_file>& result_vec
+    , std::string& error_message
+    )
+{
+    std::ifstream is(filename);
+    if(!is)
+        {
+        error_message = "'";
+        error_message.append(filename);
+        error_message.append("': No such file or directory");
+        return false;
+        }
+
+        std::string line;
+        while(std::getline(is, line))
+            {
+            // The minimal length: 32(md5sum) + 2(spaces) + 1(shortest file name)
+            const size_t minimal_line_length = 35;
+
+            // Ignore comment lines, which begin with a '#' character.
+            if(0 == line.find_first_of('#'))
+                {
+                continue;
+                }
+
+            bool format_error = true;
+            do
+                {
+                const auto space_pos = line.find_first_of(' ');
+                if(std::string::npos == line.find_first_of(' ') ||
+                    line.size() < minimal_line_length)
```

Comment:
Why combine this test with the previous one when they seem to be completely unrelated?

I'd do
```cpp
if (line.size() < minimal_line_length)
    {
    break;
    }

auto const space_pos = line.find_first_of(' ');
if(line.size() - 1 <= space_pos)
    {
    break;
    }
```

(I've changed comparison with `npos` to the one above because of another issue below).

Reply:
I fix it. But what is the issue below you are writing about?